### PR TITLE
Remove and replace depreciated ConnectionState.Connecting with ConnectionState.CatchingUp

### PR DESCRIPTION
**Pull request by RishhiB**
# Work Item
Replace remaining usages of `ConnectionState.Connecting` with `ConnectionState.CatchingUp`.

## Approach
`ConnectionState.Connecting` has been removed. Migrate all usage to `ConnectionState.CatchingUp` instead.

## Steps
The alternative for `ConnectionState.Connecting` enum value is the `ConnectionState.CatchingUp` enum value.

## Does this introduce a breaking change?

-   [X] Yes
-   [ ] No

## Other information or known dependencies

[AB#813](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/813)

### CODE REVIEW

**Review by scottn12 — APPROVED** [review]
LGTM! Let's see if any code owners have any feedback before we merge though.

File: packages/loader/container-loader/src/test/connectionStateHandler.spec.ts [hunk 1]
```diff
@@ -382,209 +382,209 @@ describe("ConnectionStateHandler Tests", () => {
 
     it("Should wait for savedEvent before moving to connected state(Client 3) when client 2 " +
         "got disconnected from connecting state", async () => {
-        client.mode = "write";
```
Comment (markfields) [inline packages/loader/container-loader/src/test/connectionStateHandler.spec.ts:385]: nit: Just collapse the string into one line on the previous line then all this indentation can be reverted.  I think I previously disabled the max-len lint rule here (I often do for test files)

**Review by markfields — APPROVED** [review]
🎂 Thanks Rishhi!  Left a few comments for places to reduce churn on whitespace changes but otherwise looks good!

File: BREAKING.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ There are a few steps you can take to write a good change note and avoid needing
 - [LocalReference class and method deprecations removed](#LocalReference-class-and-method-deprecations-removed)
 - [Remove TelemetryDataTag.PackageData](#Remove-TelemetryDataTagPackageData)
 - [Deprecate ISummaryRuntimeOptions.disableIsolatedChannels](#Deprecate-ISummaryRuntimeOptionsdisableIsolatedChannels)
+- [Remove ConnectionState.Connecting](#Remove-ConnectionState.Connecting)
```
[the remark this answers is not in the document]
Comment (scottn12) [reply BREAKING.md:26]: @markfields, yes, we should include it in the upcoming changes. From a quick check on main it looks like it was included in a note regarding a breaking change for 1.0.0, but we should probably also add an additional note to the latest upcoming section in main to give explicit warning that `ConnectionState.Connecting` will be removed in the next release.

**Pull request merged**